src/main/java/hudson/plugins/jacoco/JacocoPublisher.java

```diff
@@ -55,33 +68,54 @@
     /**
      * Variables containing the configuration set by the user.
      */
-	private final String execPattern;
-	private final String classPattern;
-	private final String sourcePattern;
-	private final String inclusionPattern;
-	private final String exclusionPattern;
-	
-    
-    private final String minimumInstructionCoverage;
-    private final String minimumBranchCoverage;
-    private final String minimumComplexityCoverage;
-    private final String minimumLineCoverage;
-    private final String minimumMethodCoverage;
-    private final String minimumClassCoverage;
-    private final String maximumInstructionCoverage;
-    private final String maximumBranchCoverage;
-    private final String maximumComplexityCoverage;
-    private final String maximumLineCoverage;
-    private final String maximumMethodCoverage;
-    private final String maximumClassCoverage;
-    private final boolean changeBuildStatus;
+    private String execPattern;
+    private String classPattern;
+    private String sourcePattern;
+    private String inclusionPattern;
+    private String exclusionPattern;
+
+    private String minimumInstructionCoverage;
+    private String minimumBranchCoverage;
+    private String minimumComplexityCoverage;
+    private String minimumLineCoverage;
+    private String minimumMethodCoverage;
+    private String minimumClassCoverage;
+    private String maximumInstructionCoverage;
+    private String maximumBranchCoverage;
+    private String maximumComplexityCoverage;
+    private String maximumLineCoverage;
+    private String maximumMethodCoverage;
+    private String maximumClassCoverage;
+    private boolean changeBuildStatus;
     
 	private static final String DIR_SEP = "\\s*,\\s*";
 
+    @DataBoundConstructor
+    public JacocoPublisher() {
+        this.execPattern = "**/**.exec";
+        this.classPattern = "**/classes";
+        this.sourcePattern = "**/src/main/java";
+        this.inclusionPattern = "";
+        this.exclusionPattern = "";
+        this.minimumInstructionCoverage = "0";
```

Comment:
I agree with you but since originally the project was using an String , I think that this change should be done in another PR for refactoring to not mix everything, and maintain this PR as the minimal changes to make this work for Pipeline.